### PR TITLE
[MM-23401] Implement switchTeam and switchChannel for SimulController

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR implements `switchChannel` and `switchTeam` actions for `SimulController`.
I took some shortcuts but most of the behaviour is there. Will improve them once I get some real feedback data from running the load-tests.

#### TIcket

https://mattermost.atlassian.net/browse/MM-23401

### CODE REVIEW

File: loadtest/control/simulcontroller/actions.go [hunk 1]
```diff
@@ -33,3 +39,108 @@ func (c *SimulController) reload(full bool) control.UserActionResponse {
...
+
+	// TODO: change this to make sure we don't pick the same channel we are
+	// currently in.
+	channel, err := u.Store().RandomChannelJoined(team.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	if current, err := u.Store().CurrentChannel(); err != nil {
+	} else if current != nil {
```
Comment (streamer45) [inline loadtest/control/simulcontroller/actions.go]: Just saw this. Need to return the error.

File: loadtest/control/simulcontroller/actions.go [hunk 2]
```diff
@@ -33,3 +39,108 @@ func (c *SimulController) reload(full bool) control.UserActionResponse {
...
+			break
+		}
+	}
+	return c.switchTeam(u)
+}
+
+func (c *SimulController) switchTeam(u user.User) control.UserActionResponse {
+	// TODO: change this to make sure we don't pick the same team we are
+	// currently in.
+	team, err := u.Store().RandomTeamJoined()
```
Comment (isacikgoz) [inline loadtest/control/simulcontroller/actions.go]: We can now update this to `RandomTeam(store.SelectMemberOf)` Also, we can add a new selection criteria, `store.CurrentlyIn` :)

Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go]: We have too :D

**Comment by streamer45** [comment]
Not sure how building got into this state. Anyway, @isacikgoz I asking for an additional review since I've added more code.

File: loadtest/control/simulcontroller/actions.go [hunk 3]
```diff
@@ -33,3 +40,105 @@ func (c *SimulController) reload(full bool) control.UserActionResponse {
...
+			}
+			c.status <- c.newInfoStatus(fmt.Sprintf("joined team %s", teamId))
+			break
+		}
+	}
+	return c.switchTeam(u)
+}
+
+func (c *SimulController) switchTeam(u user.User) control.UserActionResponse {
+	team, err := u.Store().RandomTeam(store.SelectMemberOf | store.SelectNotCurrent)
```
Comment (isacikgoz) [inline loadtest/control/simulcontroller/actions.go]: This looks very good. Thanks @agnivade for the guidance.

**Pull request merged**